contrail_api_cli/commands/relative.py

```diff
@@ -0,0 +1,53 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from ..resource import Resource
+from ..command import Command, Arg, CommandError, expand_paths
+from ..utils import printo
+
+RESOURCE_TYPE_PATH_SEPARATOR = "."
+
+
+class Relative(Command):
+    description = "Get linked resources by providing a resource type path"
+    path = Arg(help="Base resource", metavar='path')
+    resource_type_path = Arg(help="Resource types separated by '%s'" % RESOURCE_TYPE_PATH_SEPARATOR,
+                             metavar='resource_type_path')
+    show_intermediate = Arg('-l', '--show-intermediate',
+                            default=False, action="store_true",
+                            help="show intermediate resources")
+
+    def _get_next_resource(self, resource, resource_type):
+        resource.fetch()
+
+        # Try to generate the attribute corresponding to the resource type
+        if resource_type + "_refs" in resource:
+            next_resource_type = resource_type + "_refs"
+        elif resource_type + "_back_refs" in resource:
+            next_resource_type = resource_type + "_back_refs"
+        elif resource_type in resource:
+            next_resource_type = resource_type
+        else:
+            raise CommandError("Resource '%s' is not linked to resource type '%s'" % (resource.path, resource_type))
+
+        next_resource = resource.get(next_resource_type)
+        if len(next_resource) > 0:
+            return next_resource[0]
+        else:
+            return None
+
+    def __call__(self, path=None, resource_type_path=None,
+                 show_intermediate=False):
+        resources = expand_paths([path],
+                                 predicate=lambda r: isinstance(r, Resource))
+        resource = resources[0]
+
+        resource_type_paths = resource_type_path.split(
+            RESOURCE_TYPE_PATH_SEPARATOR)
+
+        for resource_type in resource_type_paths:
+            if show_intermediate:
+                printo("%s" % resource.path)
+            resource = self._get_next_resource(resource, resource_type)
+
+        return str(resource.path)
```

Comment:
Use:

from six import text_type
---
return text_type(...)

To return unicode for python2 and 3